ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -111,6 +112,15 @@ protected int getPartitions(final KafkaTopicClient kafkaTopicClient) {
     return right.getPartitions(kafkaTopicClient);
   }
 
+  @Override
+  public Stream<ColumnName> resolveSelectStar(
+      final Optional<SourceName> sourceName, final boolean valueOnly
+  ) {
+    return getSources().stream()
+        .filter(s -> !sourceName.isPresent() || sourceName.equals(s.getSourceName()))
+        .flatMap(s -> s.resolveSelectStar(sourceName, false));
```

Comment:
@rodesai made a change to the `LogicalPlanner` to alias the join columns in a projection node, prior to the join node.  There's probably scope for cleaning some of that work up a little once the dust has settled from recent changes. However: 

1). I actually probably makes sense to handle the aliases in the rewriter the engine runs post-parse, rather than in this method.
2) It's out of scope for this PR.